publishing/filterAPI/post_filter_output_event_test.go

```diff
@@ -0,0 +1,160 @@
+	}
+
+	Convey("Given a filter output does not exist", t, func() {
+		Convey("When an authorised request is made to update filter output", func() {
+			Convey("Then the request fails and returns status not found (404)", func() {
+
+				filterAPI.PUT("/filter-outputs/{filter_output_id}", filterOutputID).
+					WithHeader(serviceAuthTokenName, serviceAuthToken).
+					WithBytes([]byte(GetValidPUTFilterOutputWithCSVDownloadJSON())).
+					Expect().Status(http.StatusNotFound).Body().Contains(filterOutputNotFoundResponse)
+			})
+		})
+	})
+
+	Convey("Given an existing filter output without downloads object", t, func() {
+
+		if err := mongo.Setup(filterOutput); err != nil {
+			log.ErrorC("Unable to setup test data", err, nil)
+			os.Exit(1)
+		}
+
+		Convey("When an authorised request with invalid json body is sent to update filter output", func() {
+			Convey("Then fail to update filter output and return status bad request (400)", func() {
+
+				filterAPI.PUT("/filter-outputs/{filter_output_id}", filterOutputID).
+					WithHeader(serviceAuthTokenName, serviceAuthToken).
+					WithBytes([]byte(`{`)).
+					Expect().Status(http.StatusBadRequest).Body().Contains("Bad request - Invalid request body\n")
+			})
+		})
+
+		Convey("When an unauthorised request is made to update filter output", func() {
+			Convey("Then fail to update filter output and return status unauthorized (401)", func() {
+
+				filterAPI.PUT("/filter-outputs/{filter_output_id}", filterOutputID).
+					WithBytes([]byte(GetValidPUTFilterOutputWithCSVDownloadJSON())).
+					Expect().Status(http.StatusUnauthorized)
+			})
+		})
+
+		Convey("When an invalid authorization header is set on request to update filter output", func() {
+			Convey("Then fail to update filter output and return status unauthorized (401)", func() {
+
+				filterAPI.PUT("/filter-outputs/{filter_output_id}", filterOutputID).
+					WithHeader(serviceAuthTokenName, invalidServiceAuthToken).
+					WithBytes([]byte(GetValidPUTFilterOutputWithCSVDownloadJSON())).
+					Expect().Status(http.StatusUnauthorized)
+			})
+		})
+
+		Convey("When an authorised request is made to update filter output and json body contains dimensions or an instance id", func() {
+			Convey("Then fail to update filter output and return status forbidden (403)", func() {
+
+				filterAPI.PUT("/filter-outputs/{filter_output_id}", filterOutputID).
+					WithHeader(serviceAuthTokenName, serviceAuthToken).
+					WithBytes([]byte(GetValidPUTFilterOutputWithDimensionsJSON())).
+					Expect().Status(http.StatusForbidden).Body().Contains("Forbidden from updating the following fields: [dimensions]\n")
+
+				filterAPI.PUT("/filter-outputs/{filter_output_id}", filterOutputID).
+					WithHeader(serviceAuthTokenName, serviceAuthToken).
+					WithBytes([]byte(`{"instance_id": "1234"}`)).
+					Expect().Status(http.StatusForbidden).Body().Contains("Forbidden from updating the following fields: [instance_id]")
```

Comment:
Again no need to check response contains new line

Reply:
No newline here 😄

Reply:
sorry